vault/cluster.go

```diff
@@ -321,6 +321,11 @@ func (c *Core) startClusterListener(ctx context.Context) error {
 		// If we listened on port 0, record the port the OS gave us.
 		c.clusterAddr.Store(fmt.Sprintf("https://%s", c.getClusterListener().Addr()))
 	}
+
+	if err := c.getClusterListener().SetAdvertiseAddr(c.ClusterAddr()); err != nil {
```

Comment:
Not necessarily related to this PR in particular, but does `c.ClusterAddr()` always return the address with the scheme bit (i.e. `https://`)? If not,  `SetAdvertiseAddr` could error due to `ParseRequestURI` unable to parse that.
https://play.golang.org/p/AMT3sZmHfAT

Reply:
Yeah ClusterAddr should always be the full URL, including the scheme